spec/index.bs

```diff
@@ -596,24 +596,39 @@ Issue(gpuweb/gpuweb#296): Consider merging all read-only usages.
 Textures may consist of separate [=mipmap levels=] and [=array layers=],
 which can be used differently at any given time.
 Each such <dfn dfn>subresource</dfn> is uniquely identified by a
-[=texture=], [=mipmap level=], and
-(for {{GPUTextureDimension/2d}} textures only) [=array layer=].
+[=texture=], [=mipmap level=],
+(for {{GPUTextureDimension/2d}} textures only) [=array layer=],
+and [=aspect=].
 
-The **main usage rule** is that any [=subresource=]
-at any given time can only be in either:
+<dfn dfn>atom resource</dfn> means a whole buffer or a single subresource of a texture.
+
+The **main usage rule** is that any [=atom resource=]
+at any given time within [=usage scope=] can only be in either:
   - a combination of [=read-only usage=]s
   - a single [=mutating usage=]
 
+The only exception is that a combination of writeonly-storage-texture usages, or a combination
+of storage-buffer usages upon the same [#atom resource#] in render pass is allowed.
+
+Note:
+All invisible usages and replaced usages should be tracked, for example:
+  1) a resource binding attached to a shader stage with visiblity none,
```

Comment:
It isn't attached to a shader stage if it has visibility none (because that means no shader stages). "a resource binding with a visibility of 0"